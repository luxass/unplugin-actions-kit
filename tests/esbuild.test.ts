import { readdir, readFile } from "node:fs/promises";
import { join } from "node:path";
import { build } from "esbuild";
import { describe, expect, it } from "vitest";
import { fromFileSystem, testdir } from "vitest-testdirs";
import ActionKitPlugin from "../src/esbuild";

it("expect no `actions-kit.d.ts` file generated if plugin not in use", async () => {
	const directoryJson = await fromFileSystem("./tests/fixtures/basic");
	const testdirPath = await testdir(directoryJson);

	expect(testdirPath).toBeDefined();

	const result = await build({
		entryPoints: [join(testdirPath, "index.ts")],
		platform: "node",
		format: "cjs",
		write: false,
		bundle: false,
		minifySyntax: false,
	});

	expect(result).toBeDefined();
	expect(result.outputFiles).toBeDefined();
	const output = result.outputFiles[0];
	expect(output).toBeDefined();
	expect(output?.text).toBeDefined();
	expect(output?.text).toMatchSnapshot();

	const files = await readdir(testdirPath);
	expect(files).not.toContain("actions-kit.d.ts");
});

it("expect `actions-kit.d.ts` to be generated", async () => {
	const directoryJson = await fromFileSystem("./tests/fixtures/basic");
	const testdirPath = await testdir(directoryJson);

	expect(testdirPath).toBeDefined();

	const result = await build({
		entryPoints: [join(testdirPath, "index.ts")],
		platform: "node",
		format: "cjs",
		write: false,
		bundle: false,
		minifySyntax: false,
		plugins: [
			ActionKitPlugin({
				actionPath: join(testdirPath, "action.yaml"),
			}),
		],
	});

	const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

	expect(result).toBeDefined();
	expect(result.outputFiles).toBeDefined();
	const output = result.outputFiles[0];
	expect(output).toBeDefined();
	expect(output?.text).toBeDefined();
	expect(dtsOutput).toMatchSnapshot();
	expect(output?.text).toMatchSnapshot();
});

describe("augmentations", () => {
	it("expect only inputs to be augmented", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).not.toContain("ActionOutputName");
		expect(dtsOutput).toContain("ActionInputName");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});

	it("expect only outputs to be augmented", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).toContain("ActionOutputName");
		expect(dtsOutput).not.toContain("ActionInputName");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});

	it("expect no augmentations", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/empty");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).not.toContain("ActionOutputName");
		expect(dtsOutput).not.toContain("ActionInputName");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});
});

describe("inject", () => {
	it("expect `ACTION_INPUTS` in global scope", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
					inject: "inputs",
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).not.toContain("ActionOutputName");
		expect(dtsOutput).not.toContain("ACTION_OUTPUTS");
		expect(dtsOutput).toContain("ActionInputName");
		expect(dtsOutput).toContain("ACTION_INPUTS");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});

	it("throw if missing `inputs`", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		expect(
			async () =>
				await build({
					entryPoints: [join(testdirPath, "index.ts")],
					platform: "node",
					format: "cjs",
					write: false,
					bundle: false,
					minifySyntax: false,
					plugins: [
						ActionKitPlugin({
							actionPath: join(testdirPath, "action.yaml"),
							inject: "inputs",
						}),
					],
				}),
		).rejects.toThrow("inputs is not defined in action.yml");
	});

	it("expect `ACTION_OUTPUTS` in global scope", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
					inject: "outputs",
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).toContain("ActionOutputName");
		expect(dtsOutput).toContain("ACTION_OUTPUTS");
		expect(dtsOutput).not.toContain("ActionInputName");
		expect(dtsOutput).not.toContain("ACTION_INPUTS");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});

	it("throw if missing `outputs`", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		expect(
			async () =>
				await build({
					entryPoints: [join(testdirPath, "index.ts")],
					platform: "node",
					format: "cjs",
					write: false,
					bundle: false,
					minifySyntax: false,
					plugins: [
						ActionKitPlugin({
							actionPath: join(testdirPath, "action.yaml"),
							inject: "outputs",
						}),
					],
				}),
		).rejects.toThrow("outputs is not defined in action.yml");
	});

	it("expect `ACTION_INPUTS` & `ACTION_OUTPUTS` in global scope", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/basic");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
					inject: true,
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).toContain("ActionOutputName");
		expect(dtsOutput).toContain("ACTION_OUTPUTS");
		expect(dtsOutput).toContain("ActionInputName");
		expect(dtsOutput).toContain("ACTION_INPUTS");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});

	it("expect no change in global scope", async () => {
		const directoryJson = await fromFileSystem("./tests/fixtures/basic");
		const testdirPath = await testdir(directoryJson);

		expect(testdirPath).toBeDefined();

		const result = await build({
			entryPoints: [join(testdirPath, "index.ts")],
			platform: "node",
			format: "cjs",
			write: false,
			bundle: false,
			minifySyntax: false,
			plugins: [
				ActionKitPlugin({
					actionPath: join(testdirPath, "action.yaml"),
					inject: false,
				}),
			],
		});

		const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

		expect(result).toBeDefined();
		expect(result.outputFiles).toBeDefined();
		const output = result.outputFiles[0];
		expect(output).toBeDefined();
		expect(output?.text).toBeDefined();

		expect(dtsOutput).toContain("ActionOutputName");
		expect(dtsOutput).not.toContain("ACTION_OUTPUTS");
		expect(dtsOutput).toContain("ActionInputName");
		expect(dtsOutput).not.toContain("ACTION_INPUTS");

		expect(dtsOutput).toMatchSnapshot();
		expect(output?.text).toMatchSnapshot();
	});
});

it("custom output path", async () => {
	const directoryJson = await fromFileSystem("./tests/fixtures/basic");
	const testdirPath = await testdir(directoryJson);

	expect(testdirPath).toBeDefined();

	const result = await build({
		entryPoints: [join(testdirPath, "index.ts")],
		platform: "node",
		format: "cjs",
		write: false,
		bundle: false,
		minifySyntax: false,
		plugins: [
			ActionKitPlugin({
				actionPath: join(testdirPath, "action.yaml"),
				outputPath: join(testdirPath, "custom"),
			}),
		],
	});

	const dtsOutput = await readFile(join(testdirPath, "custom", "actions-kit.d.ts"), "utf-8");

	expect(result).toBeDefined();
	expect(result.outputFiles).toBeDefined();
	const output = result.outputFiles[0];
	expect(output).toBeDefined();
	expect(output?.text).toBeDefined();
	expect(dtsOutput).toMatchSnapshot();
	expect(output?.text).toMatchSnapshot();
});
