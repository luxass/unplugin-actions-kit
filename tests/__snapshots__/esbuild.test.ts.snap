// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`augmentations > expect no augmentations 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {


}
"
`;

exports[`augmentations > expect no augmentations 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  core.info("hello world!");
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`augmentations > expect only inputs to be augmented 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;


}
"
`;

exports[`augmentations > expect only inputs to be augmented 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`augmentations > expect only outputs to be augmented 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`augmentations > expect only outputs to be augmented 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  core.setOutput("message", "Hello, World!");
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`custom output path 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`custom output path 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
  core.setOutput("message", message);
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`expect \`actions-kit.d.ts\` to be generated 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`expect \`actions-kit.d.ts\` to be generated 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
  core.setOutput("message", message);
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`expect no \`actions-kit.d.ts\` file generated if plugin not in use 1`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
  core.setOutput("message", message);
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;

exports[`inject > expect \`ACTION_INPUTS\` & \`ACTION_OUTPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_INPUTS = {
    "type": "type",
    "who-to-greet": "who-to-greet",
  };

  export const ACTION_OUTPUTS = {
    "message": "message",
  };

}

declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect \`ACTION_INPUTS\` & \`ACTION_OUTPUTS\` in global scope 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
globalThis.ACTION_INPUTS = { "type": "type", "who-to-greet": "who-to-greet" };
globalThis.ACTION_OUTPUTS = { "message": "message" };
;
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
  core.setOutput("message", message);
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
;
"
`;

exports[`inject > expect \`ACTION_INPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_INPUTS = {
    "type": "type",
    "who-to-greet": "who-to-greet",
  };

}

declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;


}
"
`;

exports[`inject > expect \`ACTION_INPUTS\` in global scope 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
globalThis.ACTION_INPUTS = { "type": "type", "who-to-greet": "who-to-greet" };
;
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
;
"
`;

exports[`inject > expect \`ACTION_OUTPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_OUTPUTS = {
    "message": "message",
  };

}

declare module "@actions/core" {

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect \`ACTION_OUTPUTS\` in global scope 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
globalThis.ACTION_OUTPUTS = { "message": "message" };
;
async function run() {
  core.setOutput("message", "Hello, World!");
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
;
"
`;

exports[`inject > expect no change in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect no change in global scope 2`] = `
""use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var core = __toESM(require("@actions/core"), 1);
async function run() {
  const type = core.getInput("type");
  const whoToGreet = core.getInput("who-to-greet");
  const message = \`Hello \${whoToGreet}!\`;
  core.info(message);
  if (type === "error") {
    core.setFailed("This is an error!");
  }
  core.setOutput("message", message);
}
run().catch((err) => {
  console.error(err);
  core.setFailed(err.message);
});
"
`;
